### PR TITLE
Add tests for isRedeemAddress

### PR DESCRIPTION
**Pull request by erikd**
Add symmetrical Base58 encoder/decoder and deprecate fromCBORTextAddress which had no direct encoder partner.

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -4,8 +4,8 @@
 # many reviewers which often leads to nobody feeling responsible to actually
 # perform the review.
 
-crypto/  @ruhatch @dnadales
-nix/     @ruhatch @nc6
-scripts/ @ruhatch @nc6
-src/     @ruhatch @dnadales
-test/    @ruhatch @dnadales
```
Comment (intricate) [inline CODEOWNERS:11]: :disappointed: :+1:

File: cardano-ledger/src/Cardano/Chain/Common/Address.hs [hunk 2]
```diff
@@ -196,6 +197,7 @@ addressF :: Format r (Address -> r)
 addressF = build
 
 -- | A function which decodes base58-encoded 'Address'
+{-# DEPRECATED fromCBORTextAddress "Use decodeAddressBase58 instead" #-}
```
Comment (intricate) [inline cardano-ledger/src/Cardano/Chain/Common/Address.hs:200]: @erikd: is this preferred over renaming this function to `decodeAddressBase58`?

File: cardano-ledger/test/Test/Cardano/Chain/Common/Address.hs [hunk 3]
```diff
@@ -12,19 +12,27 @@ where
...
+    nm <- forAll genNetworkMagic
+    addr <- forAll (genAddressWithNM nm)
+    nm === addrNetworkMagic addr
+
+ts_prop_isRedeemAddress :: TSProperty
+ts_prop_isRedeemAddress =
+  withTestsTS 1000 . property $ do
+    addr <- forAll genAddress
+    cover 30 "Redeem Address" $ isRedeemAddress addr
+    cover 30 "Pubkey Address" $ isRedeemAddress addr
```
Comment (intricate) [inline cardano-ledger/test/Test/Cardano/Chain/Common/Address.hs]: I'm a bit confused by this as these are the exact same but with different labels. 

Is one of these intended to be `not (isRedeemAddress addr)`?

File: cardano-ledger/test/Test/Cardano/Chain/Common/CBOR.hs [hunk 4]
```diff
@@ -116,9 +119,22 @@ golden_Address4 :: Property
...
+    H.assert $ isRedeemAddress exampleAddress2
+
 ts_roundTripAddressCBOR :: TSProperty
 ts_roundTripAddressCBOR = eachOfTS 1000 genAddress roundTripsCBORBuildable
 
+ts_roundTripAddress :: TSProperty
+ts_roundTripAddress =
+  eachOfTS 1000 genAddress $ \ addr -> do
+    cover 30 "Redeem Address" $ isRedeemAddress addr
+    cover 30 "Pubkey Address" $ isRedeemAddress addr
```
Comment (intricate) [inline cardano-ledger/test/Test/Cardano/Chain/Common/CBOR.hs]: Same here. Is one of these intended to be `not (isRedeemAddress addr)`?

**Review by intricate — CHANGES_REQUESTED** [review]

File: cardano-ledger/src/Cardano/Chain/Common/Address.hs (hunk 2 above)
Comment (erikd) [reply cardano-ledger/src/Cardano/Chain/Common/Address.hs:200]: Yes, this leaves the name and produces a warning wherever it is used. The old name can be removed later.

File: cardano-ledger/test/Test/Cardano/Chain/Common/Address.hs (hunk 3 above)
Comment (erikd) [reply cardano-ledger/test/Test/Cardano/Chain/Common/Address.hs]: Yeah, I screwed up. Will fix it.

**Review by intricate — APPROVED** [review]

**Pull request merged**